docs/gadget-devel/parameters.md

```diff
@@ -30,3 +32,53 @@ params:
       defaultValue: "false"
       description: Description for the param
 ```
+
+## Parameters from metadata (gadget.yaml)
+
+Much of Inspektor Gadget's functionality is controlled by parameters and
+annotations. To give gadget authors more freedom, one can define new parameters
+that affect a group of other parameters or annotations by patching the
```

Comment:
I suggest providing more examples of what people can do with this:

affect a group of other parameters, data sources and field annotations, operator's configuration or any other field in the metadata file ...